tests/ui/option_if_let_else.rs

```diff
@@ -239,6 +239,8 @@ fn main() {
         Ok(a) => a + 1,
     };
     let _ = if let Ok(a) = res { a + 1 } else { 5 };
+    issue10729::reproduce(&None);
+    issue10729::reproduce2(&mut None);
```

Comment:
Small nit. There's no need for this as the test code is only compiled, not run.

Reply:
I did that to silence the dead code warning, but I guess I could also `#[allow]` it.

Reply:
You can also prefix names with underscores as well. That silences unused warnings.

Reply:
Decided to go with `#![allow(dead_code)]` for that module because it's also used for other unused functions in that test file, hope that's fine too